### PR TITLE
[FLINK-3292]Fix for Bug in flink-jdbc. Not all JDBC drivers supported

### PR DESCRIPTION
**Pull request by sbcd90**
Hello,

Here is the fix for issue FLINK-3292. Kindly review & merge.

Thanks & regards,
Subhobrata

### CODE REVIEW

**Comment by zentol** [comment]
is there any way to check which createStatement method is supported? what happens if you use the wrong statement?

File: flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java [hunk 1]
```diff
@@ -92,6 +94,15 @@ public void open(InputSplit ignored) throws IOException {
 		}
 	}
 
+	private Statement createStatement() throws SQLException {
+		if(resultSetType == 0 || resultSetConcurrency == 0) {
```
Comment (zentol) [inline flink-batch-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java]: this line might lead to confusion when a user only sets either the type or concurrency. 
IMO we should check with && here, require the user to set both type and concurrency at the same time, and throw a meaningful exception if only one these is 0.

**Comment by sbcd90** [comment]
Hello zentol,

I made changes according to your post. Kindly review now.

**Comment by sbcd90** [comment]
Hello zentol,

To answer your initial question,

```
is there any way to check which createStatement method is supported? what happens if you use the wrong statement?
```

I went into the Jdbc driver code & found that both methods are supported, however, with the ones defaulted in the code,
ResultSet.TYPE_SCROLL_INSENSITIVE
ResultSet.CONCUR_READ_ONLY

I got this error, 
```
java.sql.SQLException: Result set type is TYPE_FORWARD_ONLY
```

I think both variants should be supported & it is up to the api user to specify which variant they should go for. Hence, this PR. Kindly let me know if the changes in code are fine now.

**Comment by zentol** [comment]
In that case i would suggest the following:
- change the default values to TYPE_FORWARD_ONLY and CONCUR_READ_ONLY to be in line with createStatement()
-allow the user to set either one of these

**Comment by sbcd90** [comment]
Hello zentol,

Can you check the code changes now? The changes look exactly similar to what you suggest. Can you let me know if this can be merged? or do I provide a new PR?

File: flink-batch-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCOutputFormatTest.java [hunk 2]
```diff
@@ -19,10 +19,7 @@
 package org.apache.flink.api.java.io.jdbc;
 
 import java.io.IOException;
-import java.sql.Connection;
-import java.sql.DriverManager;
-import java.sql.SQLException;
-import java.sql.Statement;
+import java.sql.*;
```
Comment (zentol) [inline flink-batch-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCOutputFormatTest.java]: we don't allow star imports.

**Comment by zentol** [comment]
if you can revert the changes to the imports it's good to merge IMO.

you don't have to create a new PR. you can squash the commits if you wish, otherwise I'll do it when merging.

**Comment by sbcd90** [comment]
Hello zentol,

I have squashed all changes. Can you please merge now?

**Comment by sbcd90** [comment]
Hello zentol,

I dont see any updates from your side. Can you kindly let me know if this PR is good to be merged so that I can close the issue?

**Comment by zentol** [comment]
looks good. +1 to merge.

**Comment by StephanEwen** [comment]
Thank for the contribution and review!

Will merge this...

**Pull request closed without merging**